pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -642,15 +654,22 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
 			return fmt.Errorf("failed to find network %s", iface.Name)
 		}
 
+		ifaceType := getInterfaceType(&iface)
 		domainIface := Interface{
 			Model: &Model{
-				Type: getInterfaceType(&iface),
+				Type: ifaceType,
 			},
 			Alias: &Alias{
 				Name: iface.Name,
 			},
 		}
 
+		// if UseEmulation unset and at least one NIC model is virtio,
+		// /dev/vhost-net must be present as we should have asked for it.
+		if ifaceType == "virtio" && virtioNetProhibited {
+			return fmt.Errorf("In-kernel virtio-net device emulation '/dev/vhost-net' not present")
```

Comment:
It will, but to make it consistent with the fact that without useEmulation we should have asked for it as a requirement if virtio-net interface is needed, I thought returning error is better in that corner case.

Sent from mobile

Reply:
I understand. I just don't understand how can we get into this situation. This code is running in a virt-launcher container. A device plugin had to add `/dev/vhost-net` to it prior to the launch. So to get here the device has to disappear somehow right after the container launch. Or am I missing something? 

I wouldn't do anything special in the converter at this stage and let libvirt do the right thing.

Reply:
I get your point. It would be quite a peculiar case e.g. kernel module unloaded on the host side in the quite short window. Still possible, but we might be able to ignore it. OTOH I thought we covered similar situation for /dev/kvm in current code base. I'll check it later...

Sent from mobile

Reply:
Sorry for late comment, I meant this part:
https://github.com/kubevirt/kubevirt/blob/d5f9e2098cc6b7fbc30e165961d8715c0c02d25c/pkg/virt-launcher/virtwrap/api/converter.go#L417
I think it's better to make what `useEmulation` indicates consistent. The assumption of "the device is always there when not specifying `useEmulation`" seems reasonable for both `/dev/kvm` and `/dev/vhost-net`.

Reply:
As I see it: It can happen, although it is unlikely without a coding error and it can definitely happen if we have an error in our cluster components. Therefor I prefer to be defensive.